File: src/main/java/fi/aalto/cs/apluscourses/model/MalformedCourseConfigurationException.java
package fi.aalto.cs.apluscourses.model;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MalformedCourseConfigurationException extends Exception {

  private static final long serialVersionUID = -2707324004141394128L;
  @NotNull
  private final String pathToConfigurationFile;

  public MalformedCourseConfigurationException(@NotNull String pathToConfigurationFile,
                                               @NotNull String message,
                                               @Nullable Throwable cause) {
    super(message, cause);
    this.pathToConfigurationFile = pathToConfigurationFile;
  }

  @NotNull
  public String getPathToConfigurationFile() {
    return pathToConfigurationFile;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/model/temp/MalformedCourseConfigurationException.kt
package fi.aalto.cs.apluscourses.model.temp

import java.lang.Exception

class MalformedCourseConfigurationException(
    pathToConfigurationFile: String,
    message: String,
    cause: Throwable?
) : Exception(message, cause) {
    private val pathToConfigurationFile: String

    init {
        this.pathToConfigurationFile = pathToConfigurationFile
    }

    fun getPathToConfigurationFile(): String {
        return pathToConfigurationFile
    }
}
